<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>PhyNEST · PhyNEST</title><script data-outdated-warner src="assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.045/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.24/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href>PhyNEST</a></span></div><form class="docs-search" action="search/"><input class="docs-search-query" id="documenter-search-query" name="q" type="text" placeholder="Search docs"/></form><ul class="docs-menu"><li class="is-active"><a class="tocitem" href>PhyNEST</a><ul class="internal"><li><a class="tocitem" href="#Tutorials"><span>Tutorials</span></a></li><li><a class="tocitem" href="#Reference"><span>Reference</span></a></li><li><a class="tocitem" href="#Getting-help"><span>Getting help</span></a></li><li><a class="tocitem" href="#Functions"><span>Functions</span></a></li><li><a class="tocitem" href="#Types"><span>Types</span></a></li><li><a class="tocitem" href="#Index"><span>Index</span></a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>PhyNEST</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>PhyNEST</a></li></ul></nav><div class="docs-right"><a class="docs-edit-link" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/main/docs/src/index.md" title="Edit on GitHub"><span class="docs-icon fab"></span><span class="docs-label is-hidden-touch">Edit on GitHub</span></a><a class="docs-settings-button fas fa-cog" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-sidebar-button fa fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a></div></header><article class="content" id="documenter-page"><h1 id="PhyNEST"><a class="docs-heading-anchor" href="#PhyNEST">PhyNEST</a><a id="PhyNEST-1"></a><a class="docs-heading-anchor-permalink" href="#PhyNEST" title="Permalink"></a></h1><p>Documentation for a Julia package <strong><a href="https://github.com/sungsik-kong/PhyNEST.jl">PhyNEST</a></strong>: Phylogenetic Network Estimation using SiTe patterns.</p><hr/><h2 id="Tutorials"><a class="docs-heading-anchor" href="#Tutorials">Tutorials</a><a id="Tutorials-1"></a><a class="docs-heading-anchor-permalink" href="#Tutorials" title="Permalink"></a></h2><p>A step-by-step <a href="https://github.com/sungsik-kong/PhyNEST.jl/wiki">wiki tutorial</a> is available, that has been done for the 2023 Botany workshop.</p><h2 id="Reference"><a class="docs-heading-anchor" href="#Reference">Reference</a><a id="Reference-1"></a><a class="docs-heading-anchor-permalink" href="#Reference" title="Permalink"></a></h2><p>Sungsik Kong, David Swofford, and Laura Kubatko (2022) Inference of Phylogenetic Networks from Sequence Data using Composite Likelihood. Preprint available online on <em>BioRxiv</em> at <a href="https://doi.org/10.1101/2022.11.14.516468">https://doi.org/10.1101/2022.11.14.516468</a>.</p><h2 id="Getting-help"><a class="docs-heading-anchor" href="#Getting-help">Getting help</a><a id="Getting-help-1"></a><a class="docs-heading-anchor-permalink" href="#Getting-help" title="Permalink"></a></h2><p>Please use <a href="https://groups.google.com/g/phynest-users">google group</a> to report bugs or post questions and/or suggestions.</p><h2 id="Functions"><a class="docs-heading-anchor" href="#Functions">Functions</a><a id="Functions-1"></a><a class="docs-heading-anchor-permalink" href="#Functions" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.greet" href="#PhyNEST.greet"><code>PhyNEST.greet</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">greet()</code></pre><p>Displays a greeting with citation information. No argument needed. </p><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; greet()
Thank you for using PhyNEST: Phylogenetic Network Estimation using SiTe patterns.
Please report bugs or make suggestions to https://groups.google.com/g/phynest-users.
If you conduct an analysis using PhyNEST, please cite:
Sungsik Kong, David Swofford, and Laura Kubatko (2022) Inference of Phylogenetic Networks from Sequence Data using Composite Likelihood.
Preprint available online on BioRxiv at https://doi.org/10.1101/2022.11.14.516468.</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/miscellaneous.jl#L3-L17">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.readPhylip" href="#PhyNEST.readPhylip"><code>PhyNEST.readPhylip</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">readPhylip(inputfile::AbstractString; 
            writecsv=false::Bool,
            csvname=&quot;&quot;::AbstractString,
            showProgress=true::Bool,
            tdigits=3::Integer,
            checkpoint=false::Bool)</code></pre><p>Import, read, and parse the input phylip file. File name must be specified as string. By default, progress bar is displayed, and <code>.csv</code> and <code>.ckp</code> files are <strong>NOT</strong> produced.  See optional arguments below and modify if needed.</p><p><strong>Mandatory argument</strong></p><ul><li><code>inputfile</code>     Name of the phylip file as a string</li></ul><p><strong>Optional arguments</strong></p><ul><li><code>writecsv       (default=false)</code> A boolean arguement to allow writing site pattern frequencies in <code>.csv</code> file</li><li><code>csvname        (default=sitePatternCounts_inputfile.csv)</code> A string that will be name of the <code>.csv</code> file </li><li><code>showProgress   (default=true)</code> A boolean argument for visualizing the progress of alignment parsing</li><li><code>tdigits        (default=3)</code> Number of decimal points to record timme taken to parse the file in seconds</li><li><code>checkpoint     (default=false)</code> A boolean argument to store the Phylip object as a <code>.ckp</code> file. (Warning: <code>.ckp</code> file can be large)</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L5-L26">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.readPhylipFile" href="#PhyNEST.readPhylipFile"><code>PhyNEST.readPhylipFile</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">readPhylipFile(inputfile::AbstractString,
                writecsv::Bool,
                csvname::AbstractString,
                showProgress::Bool)</code></pre><p>Exceuted while running the function <code>readPhylip</code>.  Fills in the attributes in the <code>Phylip</code> object, except for the <code>Phylip.time</code>. </p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L50-L58">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.PhylipFileInfo" href="#PhyNEST.PhylipFileInfo"><code>PhyNEST.PhylipFileInfo</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">PhylipFileInfo(inputfile::AbstractString, 
                p::Phylip, 
                showProgress::Bool)</code></pre><p>Function that shortens?summarizes? the sequence alignment into two matrices: </p><ul><li>The one with unique sites and </li><li>Another with how many times each column in the previous matrix occurs throughout the alignment.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L79-L87">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.getUniqueQuartets" href="#PhyNEST.getUniqueQuartets"><code>PhyNEST.getUniqueQuartets</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">getUniqueQuartets(p::Phylip)</code></pre><p>Getting some combinations for four sequences (quartets). This will result in n! quartets where n is the number of sequences. I believe a better way to do this exists.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L195-L199">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.sitePatternCounts" href="#PhyNEST.sitePatternCounts"><code>PhyNEST.sitePatternCounts</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">sitePatternCounts(p::Phylip,
                ppbase::Array,
                counts::Array)</code></pre><p>Computes observed site pattern frequencies from UniqueBase and BaseCounts obtained from the function <code>PhylipFileInfo</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L220-L226">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.spRearrange" href="#PhyNEST.spRearrange"><code>PhyNEST.spRearrange</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">spRearrange(p::Phylip)</code></pre><p>Rearranges the elemetns of each unique quartet obtained using <code>getUniqueQuartets</code>, and also suffles the site pattern frequencies accordingly. This prevents 24 redundant computations, saving computation time.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L282-L286">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.show_sp" href="#PhyNEST.show_sp"><code>PhyNEST.show_sp</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">show_sp(p::Phylip)</code></pre><p>Pretty name for displaying all quartet and the corresponding site pattern frequencies on screen in the DataFrame format. It may result in excessively long table when there are many sequences in the input file.</p><p><strong>Input</strong></p><p><code>p</code>     Phylip object [mandatory]</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L390-L397">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.sitePatternsToDF" href="#PhyNEST.sitePatternsToDF"><code>PhyNEST.sitePatternsToDF</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">sitePatternsToDF(p::Phylip)</code></pre><p>Extracts the quartet and site pattern information from the Phyliip object, and reorganizes them in the DataFrame format.  This function does all hard work work for show_sp.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L403-L408">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.write_sp" href="#PhyNEST.write_sp"><code>PhyNEST.write_sp</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">write_sp(p::Phylip; 
        csvname=&quot;PhyNEST_sp&quot;::AbstractString)</code></pre><p>Pretty name for exporting all quartet and the corresponding site pattern frequencies in a .csv format.</p><p><strong>Input</strong></p><p><code>p</code>         Phylip object [mandatory] <code>csvname</code>   Filename for the .csv output can be given, otherwise we use PhyNEST_sp.csv by default.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L445-L454">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.storeCheckPoint" href="#PhyNEST.storeCheckPoint"><code>PhyNEST.storeCheckPoint</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">storeCheckPoint(p::Phylip)</code></pre><p>Stores the phylip object into a .ckp file so a user does not have to repeat the phylip parsing again if working with the  same dataset next time. By default it creates a .ckp file that has the same filename as the input phylip file. File size can get pretty large... </p><p><strong>Input</strong></p><p><code>p</code>         Phylip object [mandatory]</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L487-L496">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.readCheckPoint" href="#PhyNEST.readCheckPoint"><code>PhyNEST.readCheckPoint</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">readCheckPoint(ckpfile::AbstractString)</code></pre><p>Reads in .ckp file and creates a filled in phylip object.</p><p><strong>Input</strong></p><p><code>ckpfile</code>   Name of the checkpoint file<br/></p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/readPhylip.jl#L539-L546">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.GetTrueProbsSymm" href="#PhyNEST.GetTrueProbsSymm"><code>PhyNEST.GetTrueProbsSymm</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">GetTrueProbsSymm(myt1::Float64,
                myt2::Float64,
                myt3::Float64,
                theta::Float64)
GetTrueProbsSymm(myt1::Float64,
                myt2::Float64,
                myt3::Float64,
                theta::Float64,
                alpha::Float64)</code></pre><p>Computes true site pattern probabilities for the symmetric quartet tree: <code>((1,2),(3,4));</code>.  Three speciation times (node ages) in coalescent unit and theta must be provided; alpha is assumed to be 4/3 if unspecified.  See the manuscript and/or Chifman and Kubatko (2015)[10.1016/j.jtbi.2015.03.006] for more information.</p><p><strong>Input</strong></p><p>-<code>myt1</code>      Speciation time for the common ancestor of species 1 and 2 in coalescent unit -<code>myt2</code>      Speciation time for the common ancestor of species 3 and 4 in coalescent unit -<code>myt3</code>      Root node age in coalescent unit -<code>theta</code>     Effective population size parameter -<code>alpha</code>     (dafault=4/3)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/Probabilities.jl#L4-L25">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.GetTrueProbsAsymm" href="#PhyNEST.GetTrueProbsAsymm"><code>PhyNEST.GetTrueProbsAsymm</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">GetTrueProbsAsymm(myt1::Float64,
                myt2::Float64,
                myt3::Float64,
                theta::Float64)
GetTrueProbsAsymm(myt1::Float64,
                myt2::Float64,
                myt3::Float64,
                theta::Float64,
                alpha::Float64)</code></pre><p>Computes true site pattern probabilities for the asymmetric quartet tree: (1,(2,(3,4)));. Three speciation times (node ages) in coalescent unit and theta must be provided; alpha is assumed to be 4/3 if unspecified.  See the manuscript and/or Chifman and Kubatko (2015)[10.1016/j.jtbi.2015.03.006] for more information.</p><p><strong>Input</strong></p><p>-<code>myt1</code>      Speciation time for the common ancestor of species 3 and 4 in coalescent unit -<code>myt2</code>      Speciation time for the common ancestor of species 2 and (3,4) in coalescent unit -<code>myt3</code>      Root node age in coalescent unit -<code>theta</code>     Effective population size parameter -<code>alpha</code>     (dafault=4/3)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/Probabilities.jl#L116-L137">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.Dstat" href="#PhyNEST.Dstat"><code>PhyNEST.Dstat</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">Dstat(outgroup::String, p::Phylip;
    pval=0.05::Float64, 
    displayall=false::Bool)</code></pre><p>Conducts Patterson&#39;s D-statistic test. The result prints site pattern frequencies ABAB and ABBA used to compute  the D-statistic, Z-score, and the p-value for each quartet tested. Significance is marked with an asterisk. Function <code>showall(df)</code> can be subsequently used to show all rows.</p><p><strong>Mandatory arguments</strong></p><ul><li><code>outgroup</code>     Name of the outgroup taxa</li><li><code>p</code>   The <code>Phylip</code> object</li></ul><p><strong>Optional arguments</strong></p><ul><li><code>pval       (default=0.05)</code> Alpha level for significance</li><li><code>display_all (default=false)</code> If set as <code>true</code>, the function print test results for every quartet. By default, it only prints those quartets where signficance was found.</li></ul><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; p=readPhylip(&quot;sample_n4h1.txt&quot;)
julia&gt; df=Dstat(&quot;4&quot;,p)
Tip: if neccessary, use showall(df) function to see all the rows.
2×10 DataFrame
 Row │ outgroup  taxa1   taxa2   taxa3   ABAB   ABBA   Dstat     Zscore   pvalue   significance
     │ String    String  String  String  Int64  Int64  Float64   Float64  Float64  String
─────┼──────────────────────────────────────────────────────────────────────────────────────────
   1 │ 4         3       2       1        1427   7852  0.692424  66.6995      0.0  *
   2 │ 4         1       2       3        1427   7836  0.691892  66.5908      0.0  *

julia&gt; df=Dstat(&quot;4&quot;,p,display_all=true)
Tip: if neccessary, use showall(df) function to see all the rows.
6×10 DataFrame
Row │ outgroup  taxa1   taxa2   taxa3   ABAB   ABBA   Dstat        Zscore      pvalue    significance
    │ String    String  String  String  Int64  Int64  Float64      Float64     Float64   String
────┼─────────────────────────────────────────────────────────────────────────────────────────────────
  1 │ 4         3       1       2        7852   1427  -0.692424    -66.6995    1.0
  2 │ 4         3       2       1        1427   7852   0.692424     66.6995    0.0       *
  3 │ 4         1       3       2        7836   1427  -0.691892    -66.5908    1.0
  4 │ 4         1       2       3        1427   7836   0.691892     66.5908    0.0       *
  5 │ 4         2       3       1        7836   7852   0.00101989    0.127743  0.449176
  6 │ 4         2       1       3        7852   7836  -0.00101989   -0.127743  0.550824</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/miscellaneous.jl#L57-L99">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.showall" href="#PhyNEST.showall"><code>PhyNEST.showall</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">showall(df::DataFrame)</code></pre><p>Print all rows of the DataFrame object.    </p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/miscellaneous.jl#L160-L164">source</a></section></article><h2 id="Types"><a class="docs-heading-anchor" href="#Types">Types</a><a id="Types-1"></a><a class="docs-heading-anchor-permalink" href="#Types" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.Phylip" href="#PhyNEST.Phylip"><code>PhyNEST.Phylip</code></a> — <span class="docstring-category">Type</span></header><section><div><pre><code class="language-julia hljs">Phylip</code></pre><p>Subtype of abstract <code>INPUT</code> type with the following attributes:</p><ul><li><code>filename</code>      Name of the input phylip alignment file<br/></li><li><code>time</code>          Time taken for parsing the input in seconds<br/></li><li><code>numtaxa</code>       Number of taxa given at the first line of the input file<br/></li><li><code>seqleng</code>       Sequence length given at the first line of the input file<br/></li><li><code>nametaxa</code>      Sequence names given in the input file<br/></li><li><code>counttaxa</code>     Unique integer identifier given to each individual in the order of appearance in the input file<br/></li><li><code>allquartet</code>    All combinations of quartets for counttaxa<br/></li><li><code>index</code>         Just convert allquartet elements into arbitrary index numbers<br/></li><li><code>spcounts</code>      Arrays of 15 site pattern frequencies for each quartet in allquartet</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/5a2976788c0da50983cd76657e3e1298a1c4f0a0/src/Objects.jl#L13-L27">source</a></section></article><h2 id="Index"><a class="docs-heading-anchor" href="#Index">Index</a><a id="Index-1"></a><a class="docs-heading-anchor-permalink" href="#Index" title="Permalink"></a></h2><ul><li><a href="#PhyNEST.Phylip"><code>PhyNEST.Phylip</code></a></li><li><a href="#PhyNEST.Dstat"><code>PhyNEST.Dstat</code></a></li><li><a href="#PhyNEST.GetTrueProbsAsymm"><code>PhyNEST.GetTrueProbsAsymm</code></a></li><li><a href="#PhyNEST.GetTrueProbsSymm"><code>PhyNEST.GetTrueProbsSymm</code></a></li><li><a href="#PhyNEST.PhylipFileInfo"><code>PhyNEST.PhylipFileInfo</code></a></li><li><a href="#PhyNEST.getUniqueQuartets"><code>PhyNEST.getUniqueQuartets</code></a></li><li><a href="#PhyNEST.greet"><code>PhyNEST.greet</code></a></li><li><a href="#PhyNEST.readCheckPoint"><code>PhyNEST.readCheckPoint</code></a></li><li><a href="#PhyNEST.readPhylip"><code>PhyNEST.readPhylip</code></a></li><li><a href="#PhyNEST.readPhylipFile"><code>PhyNEST.readPhylipFile</code></a></li><li><a href="#PhyNEST.show_sp"><code>PhyNEST.show_sp</code></a></li><li><a href="#PhyNEST.showall"><code>PhyNEST.showall</code></a></li><li><a href="#PhyNEST.sitePatternCounts"><code>PhyNEST.sitePatternCounts</code></a></li><li><a href="#PhyNEST.sitePatternsToDF"><code>PhyNEST.sitePatternsToDF</code></a></li><li><a href="#PhyNEST.spRearrange"><code>PhyNEST.spRearrange</code></a></li><li><a href="#PhyNEST.storeCheckPoint"><code>PhyNEST.storeCheckPoint</code></a></li><li><a href="#PhyNEST.write_sp"><code>PhyNEST.write_sp</code></a></li></ul></article><nav class="docs-footer"><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 0.27.25 on <span class="colophon-date" title="Sunday 13 August 2023 05:38">Sunday 13 August 2023</span>. Using Julia version 1.9.2.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
