<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>PhyNEST · PhyNEST</title><script data-outdated-warner src="assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.045/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.24/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href>PhyNEST</a></span></div><form class="docs-search" action="search/"><input class="docs-search-query" id="documenter-search-query" name="q" type="text" placeholder="Search docs"/></form><ul class="docs-menu"><li class="is-active"><a class="tocitem" href>PhyNEST</a><ul class="internal"><li><a class="tocitem" href="#Tutorials"><span>Tutorials</span></a></li><li><a class="tocitem" href="#Reference"><span>Reference</span></a></li><li><a class="tocitem" href="#Getting-help"><span>Getting help</span></a></li><li><a class="tocitem" href="#Functions"><span>Functions</span></a></li><li><a class="tocitem" href="#Types"><span>Types</span></a></li><li><a class="tocitem" href="#Index"><span>Index</span></a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>PhyNEST</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>PhyNEST</a></li></ul></nav><div class="docs-right"><a class="docs-edit-link" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/main/docs/src/index.md" title="Edit on GitHub"><span class="docs-icon fab"></span><span class="docs-label is-hidden-touch">Edit on GitHub</span></a><a class="docs-settings-button fas fa-cog" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-sidebar-button fa fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a></div></header><article class="content" id="documenter-page"><h1 id="PhyNEST"><a class="docs-heading-anchor" href="#PhyNEST">PhyNEST</a><a id="PhyNEST-1"></a><a class="docs-heading-anchor-permalink" href="#PhyNEST" title="Permalink"></a></h1><p>Documentation for a Julia package <strong><a href="https://github.com/sungsik-kong/PhyNEST.jl">PhyNEST</a></strong>: Phylogenetic Network Estimation using SiTe patterns.</p><hr/><h2 id="Tutorials"><a class="docs-heading-anchor" href="#Tutorials">Tutorials</a><a id="Tutorials-1"></a><a class="docs-heading-anchor-permalink" href="#Tutorials" title="Permalink"></a></h2><p>A step-by-step <a href="https://github.com/sungsik-kong/PhyNEST.jl/wiki">wiki tutorial</a> for the major functions in <code>PhyNEST</code> is available.</p><h2 id="Reference"><a class="docs-heading-anchor" href="#Reference">Reference</a><a id="Reference-1"></a><a class="docs-heading-anchor-permalink" href="#Reference" title="Permalink"></a></h2><p>Sungsik Kong, David Swofford, and Laura Kubatko (2022) Inference of Phylogenetic Networks from Sequence Data using Composite Likelihood. Preprint available online on <em>BioRxiv</em> at <a href="https://doi.org/10.1101/2022.11.14.516468">https://doi.org/10.1101/2022.11.14.516468</a>.</p><h2 id="Getting-help"><a class="docs-heading-anchor" href="#Getting-help">Getting help</a><a id="Getting-help-1"></a><a class="docs-heading-anchor-permalink" href="#Getting-help" title="Permalink"></a></h2><p>Please use <a href="https://groups.google.com/g/phynest-users">google group</a> to report bugs or post questions and/or suggestions.</p><h2 id="Functions"><a class="docs-heading-anchor" href="#Functions">Functions</a><a id="Functions-1"></a><a class="docs-heading-anchor-permalink" href="#Functions" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.greet" href="#PhyNEST.greet"><code>PhyNEST.greet</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">greet()</code></pre><p>Displays a greeting with citation information. No argument needed. </p><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; greet()
Thank you for using PhyNEST: Phylogenetic Network Estimation using SiTe patterns.
Please report bugs or make suggestions to https://groups.google.com/g/phynest-users.
If you conduct an analysis using PhyNEST, please cite:
Sungsik Kong, David Swofford, and Laura Kubatko (2022) Inference of Phylogenetic Networks from Sequence Data using Composite Likelihood.
Preprint available online on BioRxiv at https://doi.org/10.1101/2022.11.14.516468.</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/miscellaneous.jl#L3-L17">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.readPhylip" href="#PhyNEST.readPhylip"><code>PhyNEST.readPhylip</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">readPhylip(inputfile::AbstractString; 
            writecsv=false::Bool,
            csvname=&quot;&quot;::AbstractString,
            showProgress=true::Bool,
            tdigits=3::Integer,
            checkpoint=false::Bool)</code></pre><p>Import, read, and parse the input phylip file. File name must be specified as string. By default, progress bar is displayed, and <code>.csv</code> and <code>.ckp</code> files are <strong>NOT</strong> produced.  See optional arguments below and modify if needed.</p><p><strong>Mandatory argument</strong></p><ul><li><code>inputfile</code>     Name of the phylip file as a string</li></ul><p><strong>Optional arguments</strong></p><ul><li><code>writecsv       (default=false)</code> A boolean arguement to allow writing site pattern frequencies in <code>.csv</code> file</li><li><code>csvname        (default=sitePatternCounts_inputfile.csv)</code> A string that will be name of the <code>.csv</code> file </li><li><code>showProgress   (default=true)</code> A boolean argument for visualizing the progress of alignment parsing</li><li><code>tdigits        (default=3)</code> Number of decimal points to record timme taken to parse the file in seconds</li><li><code>checkpoint     (default=false)</code> A boolean argument to store the Phylip object as a <code>.ckp</code> file. (Warning: <code>.ckp</code> file can be large)</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L5-L26">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.readPhylipFile" href="#PhyNEST.readPhylipFile"><code>PhyNEST.readPhylipFile</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">readPhylipFile(inputfile::AbstractString,
                writecsv::Bool,
                csvname::AbstractString,
                showProgress::Bool)</code></pre><p>Exceuted while running the function <code>readPhylip</code>.  Fills in the attributes in the <code>Phylip</code> object, except for the <code>Phylip.time</code>. </p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L50-L58">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.PhylipFileInfo" href="#PhyNEST.PhylipFileInfo"><code>PhyNEST.PhylipFileInfo</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">PhylipFileInfo(inputfile::AbstractString, 
                p::Phylip, 
                showProgress::Bool)</code></pre><p>Function that shortens?summarizes? the sequence alignment into two matrices: </p><ul><li>The one with unique sites and </li><li>Another with how many times each column in the previous matrix occurs throughout the alignment.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L79-L87">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.getUniqueQuartets" href="#PhyNEST.getUniqueQuartets"><code>PhyNEST.getUniqueQuartets</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">getUniqueQuartets(p::Phylip)</code></pre><p>Getting some combinations for four sequences (quartets). This will result in n! quartets where n is the number of sequences. I believe a better way to do this exists.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L195-L199">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.sitePatternCounts" href="#PhyNEST.sitePatternCounts"><code>PhyNEST.sitePatternCounts</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">sitePatternCounts(p::Phylip,
                ppbase::Array,
                counts::Array)</code></pre><p>Computes observed site pattern frequencies from UniqueBase and BaseCounts obtained from the function <code>PhylipFileInfo</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L220-L226">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.spRearrange" href="#PhyNEST.spRearrange"><code>PhyNEST.spRearrange</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">spRearrange(p::Phylip)</code></pre><p>Rearranges the elemetns of each unique quartet obtained using <code>getUniqueQuartets</code>, and also suffles the site pattern frequencies accordingly. This prevents 24 redundant computations, saving computation time.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L282-L286">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.show_sp" href="#PhyNEST.show_sp"><code>PhyNEST.show_sp</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">show_sp(p::Phylip)</code></pre><p>Pretty name for displaying all quartet and the corresponding site pattern frequencies on screen in the DataFrame format. It may result in excessively long table when there are many sequences in the input file.</p><p><strong>Mandatory argument</strong></p><ul><li><code>p</code>     Phylip object</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L390-L397">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.sitePatternsToDF" href="#PhyNEST.sitePatternsToDF"><code>PhyNEST.sitePatternsToDF</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">sitePatternsToDF(p::Phylip)</code></pre><p>Extracts the quartet and site pattern information from the Phyliip object, and reorganizes them in the DataFrame format.  This function does all hard work work for show_sp.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L403-L408">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.write_sp" href="#PhyNEST.write_sp"><code>PhyNEST.write_sp</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">write_sp(p::Phylip; 
        csvname=&quot;PhyNEST_sp&quot;::AbstractString)</code></pre><p>Pretty name for exporting all quartet and the corresponding site pattern frequencies in a .csv format.</p><p><strong>Mandatory argument</strong></p><ul><li><code>p</code>         Phylip object </li></ul><p><strong>Optional argument</strong></p><ul><li><code>csvname</code>   Filename for the .csv output can be given, otherwise we use PhyNEST_sp.csv by default.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L445-L456">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.storeCheckPoint" href="#PhyNEST.storeCheckPoint"><code>PhyNEST.storeCheckPoint</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">storeCheckPoint(p::Phylip)</code></pre><p>Stores the phylip object into a .ckp file so a user does not have to repeat the phylip parsing again if working with the  same dataset next time. By default it creates a .ckp file that has the same filename as the input phylip file. File size can get pretty large... </p><p><strong>Mandatory argument</strong></p><ul><li><code>p</code>         Phylip object</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L489-L498">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.readCheckPoint" href="#PhyNEST.readCheckPoint"><code>PhyNEST.readCheckPoint</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">readCheckPoint(ckpfile::AbstractString)</code></pre><p>Reads in .ckp file and creates a filled in phylip object.</p><p><strong>Mandatory argument</strong></p><ul><li><code>ckpfile</code>   Name of the checkpoint file</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/readPhylip.jl#L541-L548">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.GetTrueProbsSymm" href="#PhyNEST.GetTrueProbsSymm"><code>PhyNEST.GetTrueProbsSymm</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">GetTrueProbsSymm(myt1::Float64,
                myt2::Float64,
                myt3::Float64,
                theta::Float64)
GetTrueProbsSymm(myt1::Float64,
                myt2::Float64,
                myt3::Float64,
                theta::Float64,
                alpha::Float64)</code></pre><p>Computes true site pattern probabilities for the symmetric quartet tree: <code>((1,2),(3,4));</code>. The fifteen quartet site pattern probabilities are returned  in the order of:</p><ul><li>AAAA </li><li>AAAB </li><li>AABA </li><li>AABB </li><li>AABC </li><li>ABAA </li><li>ABAB </li><li>ABAC </li><li>ABBA </li><li>BAAA </li><li>ABBC </li><li>CABC </li><li>BACA </li><li>BCAA </li><li>ABCD</li></ul><p>Three speciation times (node ages) in coalescent unit and theta must be provided; alpha is assumed to be 4/3 if unspecified.  See the manuscript and/or Chifman and Kubatko (2015)[10.1016/j.jtbi.2015.03.006] for more information.</p><p><strong>Mandatory arguments</strong></p><ul><li><code>myt1</code>      Speciation time for the common ancestor of species 1 and 2 in coalescent unit</li><li><code>myt2</code>      Speciation time for the common ancestor of species 3 and 4 in coalescent unit</li><li><code>myt3</code>      Root node age in coalescent unit</li><li><code>theta</code>     Effective population size parameter</li></ul><p><strong>Optional argument</strong></p><ul><li><code>alpha (dafault=4/3)</code></li></ul><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; symprob=GetTrueProbsSymm(1.0,2.0,3.0,0.003,4/3)
15-element Vector{Float64}:
 0.2404720290824025
 0.00047873583861244776
 0.00047873583861244776
 0.0007310388098746822
 3.639464668511165e-6
 0.0007211284736310421
 6.790832193120881e-6
 1.461658030751117e-6
 6.790832193120881e-6
 0.0007211284736310421
 1.461658030751117e-6
 1.461658030751117e-6
 1.461658030751117e-6
 6.304951604760085e-6
 2.955516269578535e-8</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/Probabilities.jl#L4-L66">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.GetTrueProbsAsymm" href="#PhyNEST.GetTrueProbsAsymm"><code>PhyNEST.GetTrueProbsAsymm</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">GetTrueProbsAsymm(myt1::Float64,
                myt2::Float64,
                myt3::Float64,
                theta::Float64)
GetTrueProbsAsymm(myt1::Float64,
                myt2::Float64,
                myt3::Float64,
                theta::Float64,
                alpha::Float64)</code></pre><p>Computes true site pattern probabilities for the asymmetric quartet tree: (1,(2,(3,4)));. The fifteen quartet site pattern probabilities are returned  in the order of:</p><ul><li>AAAA </li><li>AAAB </li><li>AABA </li><li>AABB </li><li>AABC </li><li>ABAA </li><li>ABAB </li><li>ABAC </li><li>ABBA </li><li>BAAA </li><li>ABBC </li><li>CABC </li><li>BACA </li><li>BCAA </li><li>ABCD</li></ul><p>Three speciation times (node ages) in coalescent unit and theta must be provided; alpha is assumed to be 4/3 if unspecified.  See the manuscript and/or Chifman and Kubatko (2015)[10.1016/j.jtbi.2015.03.006] for more information.</p><p><strong>Mandatory arguments</strong></p><ul><li><code>myt1</code>      Speciation time for the common ancestor of species 3 and 4 in coalescent unit</li><li><code>myt2</code>      Speciation time for the common ancestor of species 2 and (3,4) in coalescent unit</li><li><code>myt3</code>      Root node age in coalescent unit</li><li><code>theta</code>     Effective population size parameter</li></ul><p><strong>Optional argument</strong></p><ul><li><code>alpha (dafault=4/3)</code></li></ul><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; asymprob=GetTrueProbsAsymm(1.0,2.0,3.0,0.003,4/3)
15-element Vector{Float64}:
 0.24055000044773364
 0.00045274538350207147
 0.00045274538350207147
 0.00027457470635933667
 1.7612660121311035e-6
 0.0006951380185206657
 3.283594866865977e-5
 1.4343273481706927e-6
 3.283594866865977e-5
 0.0011794138135323934
 2.401907811548733e-6
 1.4343273481706927e-6
 2.401907811548733e-6
 5.394333411761533e-6
 2.7254257479319977e-8</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/Probabilities.jl#L157-L219">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.GetTrueProbsAsymmTypes" href="#PhyNEST.GetTrueProbsAsymmTypes"><code>PhyNEST.GetTrueProbsAsymmTypes</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">GetTrueProbsAsymmTypes(type::Integer,
                        myt1::Float64,
                        myt2::Float64,
                        myt3::Float64,
                        theta::Float64)
GetTrueProbsAsymmTypes(type::Integer,
                        myt1::Float64,
                        myt2::Float64,
                        myt3::Float64,
                        theta::Float64,
                        alpha::Float64)</code></pre><p>Computes true site pattern probabilities for any of the four the asymmetric quartet trees: </p><ul><li>Type 1: (1,((2,3),4));</li><li>Type 2: ((1,(2,3)),4); </li><li>Type 3: (1,(2,(3,4))); </li><li>Type 4: (((1,2),3),4);</li></ul><p>The fifteen quartet site pattern probabilities are returned in the order of:</p><ul><li>AAAA </li><li>AAAB </li><li>AABA </li><li>AABB </li><li>AABC </li><li>ABAA </li><li>ABAB </li><li>ABAC </li><li>ABBA </li><li>BAAA </li><li>ABBC </li><li>CABC </li><li>BACA </li><li>BCAA </li><li>ABCD</li></ul><p>Three speciation times (node ages) in coalescent unit and theta must be provided; alpha is assumed to be 4/3 if unspecified.  See the manuscript and/or Chifman and Kubatko (2015)[10.1016/j.jtbi.2015.03.006] for more information.</p><p><strong>Mandatory arguments</strong></p><ul><li><code>type</code>      Type of the asymmetric quartet in interest. See above.</li><li><code>myt1</code>      Speciation time for the common ancestor of species 3 and 4 in coalescent unit</li><li><code>myt2</code>      Speciation time for the common ancestor of species 2 and (3,4) in coalescent unit</li><li><code>myt3</code>      Root node age in coalescent unit</li><li><code>theta</code>     Effective population size parameter</li></ul><p><strong>Optional argument</strong></p><ul><li><code>alpha (dafault=4/3)</code></li></ul><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; GetTrueProbsAsymmTypes(1,1.0,2.0,3.0,0.003,4/3)
15-element Vector{Float64}:
 0.24055000044773364
 0.0006951380185206657
 0.00045274538350207147
 3.283594866865977e-5
 1.4343273481706927e-6
 0.00045274538350207147
 3.283594866865977e-5
 1.4343273481706927e-6
 0.00027457470635933667
 0.0011794138135323934
 5.394333411761533e-6
 1.7612660121311035e-6
 2.401907811548733e-6
 2.401907811548733e-6
 2.7254257479319977e-8

julia&gt; GetTrueProbsAsymmTypes(2,1.0,2.0,3.0,0.003,4/3)
15-element Vector{Float64}:
 0.24055000044773364
 0.0011794138135323934
 0.00045274538350207147
 3.283594866865977e-5
 2.401907811548733e-6
 0.00045274538350207147
 3.283594866865977e-5
 2.401907811548733e-6
 0.00027457470635933667
 0.0006951380185206657
 5.394333411761533e-6
 1.7612660121311035e-6
 1.4343273481706927e-6
 1.4343273481706927e-6
 2.7254257479319977e-8

julia&gt; GetTrueProbsAsymmTypes(3,1.0,2.0,3.0,0.003,4/3)
15-element Vector{Float64}:
 0.24055000044773364
 0.00045274538350207147
 0.00045274538350207147
 0.00027457470635933667
 1.7612660121311035e-6
 0.0006951380185206657
 3.283594866865977e-5
 1.4343273481706927e-6
 3.283594866865977e-5
 0.0011794138135323934
 2.401907811548733e-6
 1.4343273481706927e-6
 2.401907811548733e-6
 5.394333411761533e-6
 2.7254257479319977e-8

julia&gt; GetTrueProbsAsymmTypes(4,1.0,2.0,3.0,0.003,4/3)
15-element Vector{Float64}:
 0.24055000044773364
 0.0011794138135323934
 0.0006951380185206657
 0.00027457470635933667
 5.394333411761533e-6
 0.00045274538350207147
 3.283594866865977e-5
 2.401907811548733e-6
 3.283594866865977e-5
 0.00045274538350207147
 2.401907811548733e-6
 1.4343273481706927e-6
 1.4343273481706927e-6
 1.7612660121311035e-6
 2.7254257479319977e-8</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/Probabilities.jl#L393-L518">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.sim_sp_freq" href="#PhyNEST.sim_sp_freq"><code>PhyNEST.sim_sp_freq</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">sim_sp_freq(type::Integer,
            myt1::Float64,
            myt2::Float64,
            myt3::Float64,
            theta::Float64)
sim_sp_freq(type::Integer,
            myt1::Float64,
            myt2::Float64,
            myt3::Float64,
            theta::Float64,
            alpha::Float64,
            length::Integer)</code></pre><p>Generates site pattern frequencies for the five possible quartet topologies modeled as  a multinomial random variable under the assumption that the observed sites are independent,  conditional on the species tree. The five topologies are:</p><ul><li>Type 0: ((1,2),(3,4));</li><li>Type 1: (1,((2,3),4));</li><li>Type 2: ((1,(2,3)),4); </li><li>Type 3: (1,(2,(3,4))); </li><li>Type 4: (((1,2),3),4);</li></ul><p>The fifteen quartet site pattern frequencies are returned in the order of:</p><ul><li>AAAA </li><li>AAAB </li><li>AABA </li><li>AABB </li><li>AABC </li><li>ABAA </li><li>ABAB </li><li>ABAC </li><li>ABBA </li><li>BAAA </li><li>ABBC </li><li>CABC </li><li>BACA </li><li>BCAA </li><li>ABCD</li></ul><p><strong>Mandatory arguments</strong></p><ul><li><code>type</code>      Type of the asymmetric quartet in interest. See above.</li><li><code>myt1</code>      Speciation time for the common ancestor of species 3 and 4 in coalescent unit</li><li><code>myt2</code>      Speciation time for the common ancestor of species 2 and (3,4) in coalescent unit</li><li><code>myt3</code>      Root node age in coalescent unit</li><li><code>theta</code>     Effective population size parameter</li></ul><p><strong>Optional argument</strong></p><ul><li><code>alpha (dafault=4/3)</code></li><li><code>length</code> (default=1000000) Alignment length</li></ul><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; sim_sp_freq(1,1.0,2.0,3.0,0.003,4/3,50000)
15-element Vector{Int64}:
 48073
   411
   261
    14
     6
   285
    33
     1
   171
   729
     9
     1
     2
     4
     0
julia&gt; sim_sp_freq(2,1.0,2.0,3.0,0.003)
15-element Vector{Int64}:
 962053
  14160
   5460
    410
     58
   5405
    380
     57
   3372
   8434
    122
     32
     32
     24
      1     </code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/Probabilities.jl#L546-L637">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.phyne!" href="#PhyNEST.phyne!"><code>PhyNEST.phyne!</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">phyne!(starting_topology::HybridNetwork,p::Phylip,outgroup::String;
        hmax=1::Integer,
        maximum_number_of_steps=250000::Integer,
        maximum_number_of_failures=100::Integer,
        number_of_itera=1000::Integer,
        number_of_runs=10::Integer,
        do_hill_climbing=true::Bool,
        number_of_burn_in=25::Integer,
        k=10::Integer,
        cons=0.5::Float64,
        alph=0.5::Float64,
        filename=&quot;&quot;::AbstractString)</code></pre><p><code>phyne!</code> is fine. <code>phyne!</code> executes function <code>initiate_search(args)</code>.</p><p>Estimate the species network (or tree if <code>hmax=0</code>) using maximum composite likelihood. The search begins from the <code>starting_topology</code> which can be either estimated or randomly generated. Starting topology can be either tree or a network with <code>&lt;=hamx</code>. Outgroup taxon  must be specified to root the network. </p><p><strong>Mandatory arguments</strong></p><ul><li><code>starting_topology</code>       Starting topology in <code>HybridNetwork</code> object created using the function <code>readTopology</code> </li><li><code>p</code>                       Sequence alignment parsed in <code>Phylip</code> object using the function <code>readPhylip</code></li><li><code>outgroup</code>                Name of the outgroup taxa</li></ul><p><strong>Optional arguments</strong></p><p><strong>Generic</strong></p><ul><li><code>hmax (dafault=1)</code>                            Maximum number of hybridizations to be included in the final network</li><li><code>do_hill_climbing (default=true)</code>             When <code>=true</code>, network is searched using hill climbing and when <code>=false</code>, it searches using simulated annealing</li><li><code>number_of_runs (default=10)</code>                 Number of independent runs</li></ul><p><strong>Network space search</strong></p><ul><li><code>maximum_number_of_steps (default=250000)</code>    Maximum number of steps before the search terminates</li><li><code>maximum_number_of_failures (default=100)</code>    Maximum number of consecutive failures (i.e., rejecting the proposed topology) before the search teminates</li></ul><p><strong>Optimization</strong></p><ul><li><code>number_of_itera (default=1000)</code></li></ul><p><strong>For simulated annealing</strong></p><ul><li><code>number_of_burn_in (default=25)</code>              </li><li><code>k (default=10)</code>                              Specifies the number of best networks to be stored at each run</li><li><code>cons (default=0.5)</code>                          </li><li><code>alph (default=0.5)</code></li></ul><p><strong>Output</strong></p><ul><li><code>filename (default=&quot;&quot;)</code> Specifies the name of the output file. If unspecified, it will use <code>PhyNEST_hc</code> or <code>PhyNEST_sa</code> depending on the heuristic method applied.</li><li>The best network estimated throughout the entire runs is written in the extended Newick format and stored in file with an extension <code>.out</code>. </li></ul><p>First line in the file is readable by <code>PhyloNetworks</code> that can be visualized using <code>PhyloPlots</code>. Third (last) line is the identical network but readable by <code>DendroScope</code>. </p><ul><li>Specifics of the network search from each independent run is stored in the file with an extension <code>.log</code>. </li></ul><p>It summarizes how many of each network &#39;moves&#39; were made during the search, the reason for terminating the search  (e.g., researched maximum number of steps, reached maximum number of consecutive failures, or else), and  the final network estimated and its composite likelihood. When <code>do_hill_climbing=true</code>, a single best network is selected for each run, but when <code>do_hill_climbing=false</code> it prints <code>k</code> best networks visited.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/Searches.jl#L651-L705">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.Dstat" href="#PhyNEST.Dstat"><code>PhyNEST.Dstat</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">Dstat(p::Phylip, outgroup::String;
    alpha=0.05::Float64, 
    displayall=false::Bool,
    writecsv=false::Bool, 
    filename=&quot;&quot;::AbstractString)</code></pre><p>Conducts Patterson&#39;s D-statistic test. The result prints site pattern frequencies ABAB and ABBA used to compute  the D-statistic, Z-score, and the p-value for each quartet tested. Significance is marked with an asterisk. Function <code>showall(df)</code> can be subsequently used to show all rows.</p><p><strong>Mandatory arguments</strong></p><ul><li><code>p</code>   The <code>Phylip</code> object</li><li><code>outgroup</code>     Name of the outgroup taxa</li></ul><p><strong>Optional arguments</strong></p><ul><li><code>alpha       (default=0.05)</code> Alpha level for significance</li><li><code>display_all (default=false)</code> If set as <code>true</code>, the function print test results for every quartet. By default, it only prints those quartets where the signficance was found.</li><li><code>writecsv (default=false)</code> If <code>true</code>, the result is stored in <code>.csv</code> file in the working directory</li><li><code>filename</code> Specifies <code>.csv</code> file name if <code>writecsv=true</code>. If unspecified, the result is stored as <code>Dstat-out.csv</code></li></ul><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; p=readPhylip(&quot;sample_n4h1.txt&quot;)
julia&gt; df=Dstat(p,&quot;4&quot;)
Tip: if neccessary, use showall(df) function to see all the rows.
2×10 DataFrame
 Row │ outgroup  taxa1   taxa2   taxa3   ABAB   ABBA   Dstat     Zscore   pvalue   significance
     │ String    String  String  String  Int64  Int64  Float64   Float64  Float64  String
─────┼──────────────────────────────────────────────────────────────────────────────────────────
   1 │ 4         3       2       1        1427   7852  0.692424  66.6995      0.0  *
   2 │ 4         1       2       3        1427   7836  0.691892  66.5908      0.0  *

julia&gt; df=Dstat(p,&quot;4&quot;,display_all=true)
Tip: if neccessary, use showall(df) function to see all the rows.
6×10 DataFrame
Row │ outgroup  taxa1   taxa2   taxa3   ABAB   ABBA   Dstat        Zscore      pvalue    significance
    │ String    String  String  String  Int64  Int64  Float64      Float64     Float64   String
────┼─────────────────────────────────────────────────────────────────────────────────────────────────
  1 │ 4         3       1       2        7852   1427  -0.692424    -66.6995    1.0
  2 │ 4         3       2       1        1427   7852   0.692424     66.6995    0.0       *
  3 │ 4         1       3       2        7836   1427  -0.691892    -66.5908    1.0
  4 │ 4         1       2       3        1427   7836   0.691892     66.5908    0.0       *
  5 │ 4         2       3       1        7836   7852   0.00101989    0.127743  0.449176
  6 │ 4         2       1       3        7852   7836  -0.00101989   -0.127743  0.550824</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/miscellaneous.jl#L57-L103">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.showallDF" href="#PhyNEST.showallDF"><code>PhyNEST.showallDF</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">showallDF(df::DataFrame)</code></pre><p>Print all rows of the DataFrame object using the package <code>CSV</code>.    </p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/miscellaneous.jl#L175-L179">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.HyDe" href="#PhyNEST.HyDe"><code>PhyNEST.HyDe</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">HyDe(p::Phylip, outgroup::AbstractString; 
    alpha=0.05::Float64, 
    display_all=true::Bool, #filter
    map=&quot;&quot;::AbstractString,
    writecsv=false::Bool, 
    filename=&quot;&quot;::AbstractString</code></pre><p>Conducts HyDe: Hybrid Detection using Phylogenetic invariants. See Blischak et al., (2018) (https://doi.org/10.1093/sysbio/syy023) and the manual (https://hybridization-detection.readthedocs.io) for more information.  This function replicates <code>run_hyde.py</code> in the original python package. The map file, a two-column table with individual names in the first column and the name of the population that it belongs to in the second column (see example below) is not necessary to start the analysis. If the map file is not provided, each sequnece in the data is assumed to represent distinct species. </p><p>Map file is a simple text file where each line contains the name of the sequencea and the assignment of the sequence to a species, delimited by a tab. See an example below.</p><p><strong>Mandatory arguments</strong></p><ul><li><code>p</code>   The <code>Phylip</code> object        </li><li><code>outgroup</code>     Name of the outgroup taxa. Even when there are muliple outgroup individuals, but if their assignment to the species is provided in the <code>map</code> file, simply specify one of the outgroup species as listed in the alignment.</li></ul><p><strong>Optional arguments</strong></p><ul><li><code>alpha       (default=0.05)</code> Alpha level for significance</li><li><code>display_all (default=false)</code> If set as <code>true</code>, results are also filtered based on whether there is significant evidence for hybridization.</li><li><code>map (default=no map file)</code>   Specify a map file, if available. </li><li><code>writecsv (default=false)</code> If <code>true</code>, the result is stored in <code>.csv</code> file in the working directory</li><li><code>filename</code> Specifies <code>.csv</code> file name if <code>writecsv=true</code>. If unspecified, the result is stored as <code>HyDe-out.csv</code></li></ul><p><strong>Example</strong></p><pre><code class="language- hljs">julia&gt; HyDe(p,&quot;5&quot;,display_all=true)
24×11 DataFrame
 Row │ outgroup  P1      Hybrid  P2      AABB   ABAB   ABBA   Gamma         Zscore         Pvalue    significance
     │ String    String  String  String  Int64  Int64  Int64  Float64       Float64        Float64   String
─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ 5         4       3       1       18168   2573   2611    0.00243076       0.528406  0.298609
   2 │ 5         4       3       2       23742   2022   2064    0.00192997       0.657666  0.255376
   3 │ 5         4       1       2       23703   2069   2069    0.0         -99999.9       1.0
   4 │ 5         3       1       2        8005   8057   1991    0.9915          -0.412067  0.659855
   5 │ 5         4       1       3       18168   2611   2573   -0.00244861  -99999.9       1.0
   6 │ 5         3       4       1        2573  18168   2611    0.49939       -216.327     1.0
   7 │ 5         3       1       4        2573   2611  18168    1.00245         -0.527118  0.700944
   8 │ 5         1       4       3        2611  18168   2573    0.50061       -216.327     1.0
   9 │ 5         1       3       4        2611   2573  18168    0.997569         0.528406  0.298609
  10 │ 5         4       2       3       23742   2064   2022   -0.00194121  -99999.9       1.0
  11 │ 5         3       4       2        2022  23742   2064    0.499516      -339.45      1.0
  12 │ 5         3       2       4        2022   2064  23742    1.00194         -0.656395  0.744215
  13 │ 5         2       4       3        2064  23742   2022    0.500484      -339.45      1.0
  14 │ 5         2       3       4        2064   2022  23742    0.99807          0.657666  0.255376
  15 │ 5         4       2       1       23703   2069   2069    0.0         -99999.9       1.0
  16 │ 5         1       4       2        2069  23703   2069    0.5           -336.307     1.0
  17 │ 5         1       2       4        2069   2069  23703  NaN                0.0       0.5
  18 │ 5         2       4       1        2069  23703   2069    0.5           -336.307     1.0
  19 │ 5         2       1       4        2069   2069  23703  NaN                0.0       0.5
  20 │ 5         3       2       1        8005   1991   8057    0.502152        47.6571    0.0       *
  21 │ 5         1       3       2        8057   8005   1991    1.00872     -99999.9       1.0
  22 │ 5         1       2       3        8057   1991   8005    0.497848        47.6571    0.0       *
  23 │ 5         2       3       1        1991   8005   8057   -0.00872191      -0.408534  0.658559
  24 │ 5         2       1       3        1991   8057   8005    0.00849951      -0.412067  0.659855

julia&gt; HyDe(p,&quot;5&quot;,display_all=false)
2×11 DataFrame
 Row │ outgroup  P1      Hybrid  P2      AABB   ABAB   ABBA   Gamma     Zscore   Pvalue   significance
     │ String    String  String  String  Int64  Int64  Int64  Float64   Float64  Float64  String
─────┼─────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ 5         3       2       1        8005   1991   8057  0.502152  47.6571      0.0  *
   2 │ 5         1       2       3        8057   1991   8005  0.497848  47.6571      0.0  *

julia&gt; HyDe(p,&quot;5&quot;,map=&quot;map.txt&quot;,display_all=false)
Map file [map.txt] provided.
2×11 DataFrame
 Row │ outgroup  P1      Hybrid  P2      AABB   ABAB   ABBA   Gamma     Zscore   Pvalue   significance
     │ String    String  String  String  Int64  Int64  Int64  Float64   Float64  Float64  String
─────┼─────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ sp5out    sp3     sp2     sp1     15841   3418  15909  0.501365  49.4337      0.0  *
   2 │ sp5out    sp1     sp2     sp3     15909   3418  15841  0.498635  49.4337      0.0  *</code></pre><p>where the <code>map.txt</code> used in above example is:</p><pre><code class="nohighlight hljs">shell&gt; cat map.txt
5	sp5out
4	sp5out
3	sp3
1	sp1
2	sp2</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/miscellaneous.jl#L185-L271">source</a></section></article><h2 id="Types"><a class="docs-heading-anchor" href="#Types">Types</a><a id="Types-1"></a><a class="docs-heading-anchor-permalink" href="#Types" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.Phylip" href="#PhyNEST.Phylip"><code>PhyNEST.Phylip</code></a> — <span class="docstring-category">Type</span></header><section><div><pre><code class="language-julia hljs">Phylip</code></pre><p>Subtype of abstract <code>INPUT</code> type with the following attributes:</p><ul><li><code>filename</code>      Name of the input phylip alignment file</li><li><code>time</code>          Time taken for parsing the input in seconds</li><li><code>numtaxa</code>       Number of taxa given at the first line of the input file</li><li><code>seqleng</code>       Sequence length given at the first line of the input file</li><li><code>nametaxa</code>      Sequence names given in the input file</li><li><code>counttaxa</code>     Unique integer identifier given to each individual in the order of appearance in the input file</li><li><code>allquartet</code>    All combinations of quartets for counttaxa</li><li><code>index</code>         Just convert allquartet elements into arbitrary index numbers</li><li><code>spcounts</code>      Arrays of 15 site pattern frequencies for each quartet in allquartet</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/Objects.jl#L13-L27">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.quartets" href="#PhyNEST.quartets"><code>PhyNEST.quartets</code></a> — <span class="docstring-category">Type</span></header><section><div><pre><code class="language-julia hljs">quartets</code></pre><p>Subtype of abstract <code>Quartet</code> type with the following attributes:</p><ul><li><code>number</code>        List of individuals quartets in the order of 1,2,3,4</li><li><code>displayed_tree</code>nth displayed tree that the quartet was extracted from</li><li><code>quartet</code>       List of quartets in the order of 1,2,3,4 using the leaf numbers in HybridNetwork</li><li><code>tquartet</code>      List of quartets in the order of 1,2,3,4 using the leaf numbers in Phylip</li><li><code>gamma</code>         Inhertiance probability information provided in HybridNetwork</li><li><code>mspcountsNET</code>  We can directly use this counts for likelihood calculation.</li><li><code>mrca</code>          List of common ancesters of two taxa in the order of 1 and 2 (12),13,14,23,24,and 34</li><li><code>ntau</code>          Unique number of the taus in the tau used in branchlengths</li><li><code>momestlength</code>  Branch length identified for each quartet given the mspcounts using moment estimator</li><li><code>average_mom_est_bl</code>         Branch length that is averaged for the entire tree/network - Will get filled later because theta is required</li><li><code>symtype</code>       Type of each quartet. It can be either symmetric (type 0) or asymmetric. Asymmetric quartets have four possible topologies:<ul><li>Type 0: ((1,2),(3,4));</li><li>Type 1: (1,((2,3),4));</li><li>Type 2: ((1,(2,3)),4); </li><li>Type 3: (1,(2,(3,4))); </li><li>Type 4: (((1,2),3),4);</li></ul></li><li><code>logLik</code>        It&#39;s just a negative likelihood for the quartet in interest</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/Objects.jl#L61-L83">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.Network" href="#PhyNEST.Network"><code>PhyNEST.Network</code></a> — <span class="docstring-category">Type</span></header><section><div><p>Network</p><p>Subtype of abstract <code>Quartet</code> type with the following attributes:</p><ul><li><code>leafname</code>    Name of the leaves in the order of appearance in the input newick<br/></li><li><code>leafnumber</code>  Assigned number for each leaf when reading in the input newick<br/></li><li><code>gamma</code>       Gamma as appears on the newick, although irrelevant for our computation<br/></li><li><code>theta</code>       Estimated theta using the site pattern frequencies. If Network object is<br/>              created without Phylip, theta is not estimated byt default theta=0.001 is displayed.<br/></li><li><code>quartet</code>     An array of all quartets in the given topology</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/b30fc04aa14aab29709f28fcd171cd7d3083b176/src/Objects.jl#L101-L112">source</a></section></article><h2 id="Index"><a class="docs-heading-anchor" href="#Index">Index</a><a id="Index-1"></a><a class="docs-heading-anchor-permalink" href="#Index" title="Permalink"></a></h2><ul><li><a href="#PhyNEST.Network"><code>PhyNEST.Network</code></a></li><li><a href="#PhyNEST.Phylip"><code>PhyNEST.Phylip</code></a></li><li><a href="#PhyNEST.quartets"><code>PhyNEST.quartets</code></a></li><li><a href="#PhyNEST.Dstat"><code>PhyNEST.Dstat</code></a></li><li><a href="#PhyNEST.GetTrueProbsAsymm"><code>PhyNEST.GetTrueProbsAsymm</code></a></li><li><a href="#PhyNEST.GetTrueProbsAsymmTypes"><code>PhyNEST.GetTrueProbsAsymmTypes</code></a></li><li><a href="#PhyNEST.GetTrueProbsSymm"><code>PhyNEST.GetTrueProbsSymm</code></a></li><li><a href="#PhyNEST.HyDe"><code>PhyNEST.HyDe</code></a></li><li><a href="#PhyNEST.PhylipFileInfo"><code>PhyNEST.PhylipFileInfo</code></a></li><li><a href="#PhyNEST.getUniqueQuartets"><code>PhyNEST.getUniqueQuartets</code></a></li><li><a href="#PhyNEST.greet"><code>PhyNEST.greet</code></a></li><li><a href="#PhyNEST.phyne!"><code>PhyNEST.phyne!</code></a></li><li><a href="#PhyNEST.readCheckPoint"><code>PhyNEST.readCheckPoint</code></a></li><li><a href="#PhyNEST.readPhylip"><code>PhyNEST.readPhylip</code></a></li><li><a href="#PhyNEST.readPhylipFile"><code>PhyNEST.readPhylipFile</code></a></li><li><a href="#PhyNEST.show_sp"><code>PhyNEST.show_sp</code></a></li><li><a href="#PhyNEST.showallDF"><code>PhyNEST.showallDF</code></a></li><li><a href="#PhyNEST.sim_sp_freq"><code>PhyNEST.sim_sp_freq</code></a></li><li><a href="#PhyNEST.sitePatternCounts"><code>PhyNEST.sitePatternCounts</code></a></li><li><a href="#PhyNEST.sitePatternsToDF"><code>PhyNEST.sitePatternsToDF</code></a></li><li><a href="#PhyNEST.spRearrange"><code>PhyNEST.spRearrange</code></a></li><li><a href="#PhyNEST.storeCheckPoint"><code>PhyNEST.storeCheckPoint</code></a></li><li><a href="#PhyNEST.write_sp"><code>PhyNEST.write_sp</code></a></li></ul></article><nav class="docs-footer"><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 0.27.25 on <span class="colophon-date" title="Saturday 26 August 2023 19:11">Saturday 26 August 2023</span>. Using Julia version 1.9.3.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
