<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>PhyNEST.jl · PhyNEST.jl</title><script data-outdated-warner src="assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.045/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.24/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><a class="docs-logo" href><img src="assets/logo.png" alt="PhyNEST.jl logo"/></a><div class="docs-package-name"><span class="docs-autofit"><a href>PhyNEST.jl</a></span></div><form class="docs-search" action="search/"><input class="docs-search-query" id="documenter-search-query" name="q" type="text" placeholder="Search docs"/></form><ul class="docs-menu"><li class="is-active"><a class="tocitem" href>PhyNEST.jl</a><ul class="internal"><li><a class="tocitem" href="#References"><span>References</span></a></li><li><a class="tocitem" href="#Getting-help"><span>Getting help</span></a></li><li><a class="tocitem" href="#Manual"><span>Manual</span></a></li><li><a class="tocitem" href="#Functions"><span>Functions</span></a></li><li><a class="tocitem" href="#Types"><span>Types</span></a></li><li><a class="tocitem" href="#Index"><span>Index</span></a></li></ul></li><li><span class="tocitem">Manual</span><ul><li><a class="tocitem" href="manual/installation/">Installation</a></li><li><a class="tocitem" href="manual/input/">Input</a></li><li><a class="tocitem" href="manual/networkest/">Network estimation</a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>PhyNEST.jl</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>PhyNEST.jl</a></li></ul></nav><div class="docs-right"><a class="docs-edit-link" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/main/docs/src/index.md" title="Edit on GitHub"><span class="docs-icon fab"></span><span class="docs-label is-hidden-touch">Edit on GitHub</span></a><a class="docs-settings-button fas fa-cog" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-sidebar-button fa fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a></div></header><article class="content" id="documenter-page"><h1 id="PhyNEST.jl"><a class="docs-heading-anchor" href="#PhyNEST.jl">PhyNEST.jl</a><a id="PhyNEST.jl-1"></a><a class="docs-heading-anchor-permalink" href="#PhyNEST.jl" title="Permalink"></a></h1><p>Documentation for a Julia package <strong><a href="https://github.com/sungsik-kong/PhyNEST.jl">PhyNEST.jl</a>: Phylogenetic Network Estimation using SiTe patterns</strong>.</p><hr/><h2 id="References"><a class="docs-heading-anchor" href="#References">References</a><a id="References-1"></a><a class="docs-heading-anchor-permalink" href="#References" title="Permalink"></a></h2><p>Sungsik Kong, David Swofford, and Laura Kubatko (2022) Inference of Phylogenetic Networks from Sequence Data using Composite Likelihood. Preprint available online on <em>BioRxiv</em> at <a href="https://doi.org/10.1101/2022.11.14.516468">https://doi.org/10.1101/2022.11.14.516468</a>.</p><h2 id="Getting-help"><a class="docs-heading-anchor" href="#Getting-help">Getting help</a><a id="Getting-help-1"></a><a class="docs-heading-anchor-permalink" href="#Getting-help" title="Permalink"></a></h2><p>Please use <a href="https://groups.google.com/g/phynest-users">google group</a> to report bugs or post questions and/or suggestions.</p><h2 id="Manual"><a class="docs-heading-anchor" href="#Manual">Manual</a><a id="Manual-1"></a><a class="docs-heading-anchor-permalink" href="#Manual" title="Permalink"></a></h2><ul><li><a href="manual/installation/#Installation">Installation</a></li><li class="no-marker"><ul><li><a href="manual/installation/#Julia-installation">Julia installation</a></li><li><a href="manual/installation/#Installation-of-the-package-PhyNEST">Installation of the package PhyNEST</a></li><li><a href="manual/installation/#Test-example">Test example</a></li></ul></li><li><a href="manual/input/#Input">Input</a></li><li class="no-marker"><ul><li><a href="manual/input/#Parsing-DNA-alignment-data">Parsing DNA alignment data</a></li><li><a href="manual/input/#Reading-.ckp-file">Reading <code>.ckp</code> file</a></li></ul></li><li><a href="manual/networkest/#Network-estimation">Network estimation</a></li><li class="no-marker"><ul><li><a href="manual/networkest/#Getting-ready-for-the-network-estimation">Getting ready for the network estimation</a></li><li><a href="manual/networkest/#Hill-climbing">Hill climbing</a></li><li><a href="manual/networkest/#Simulated-annealing">Simulated annealing</a></li></ul></li></ul><h2 id="Functions"><a class="docs-heading-anchor" href="#Functions">Functions</a><a id="Functions-1"></a><a class="docs-heading-anchor-permalink" href="#Functions" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.greet" href="#PhyNEST.greet"><code>PhyNEST.greet</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">greet()</code></pre><p>Displays a simple greet message with citation information. No input argument is needed. </p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/miscellaneous.jl#L14-L18">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.readPhylip" href="#PhyNEST.readPhylip"><code>PhyNEST.readPhylip</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">readPhylip(inputfile::AbstractString)</code></pre><p>Import, read, and parse the input phylip file. File name must be specified as string. <br/>By default, .csv and .ckp files are <strong>NOT</strong> produced. See below for the optional arguments.\ </p><p>To save the observed quartet site pattern frequencies in a .csv file, set the optional argument<br/><code>writecsv=true</code>. The .csv file will be stored in the working directory, with a filename <br/><code>&quot;sitePatternCounts_inputfile.csv&quot;</code>. If a user would like to modify the filename, it can be done<br/>by providing preferred name using the optional argument <code>csvname=&quot;preferred_file_name.csv&quot;</code>. <br/></p><p><strong>Input</strong></p><p><strong>Mandatory</strong></p><ul><li><code>inputfile</code>     Name of the phylip file as a String [mandatory]\ </li></ul><p><strong>Optional</strong></p><ul><li><code>writecsv       (default=false)</code> A Boolean arguement to write site pattern frequencies in CSV file<br/></li><li><code>csvname        (default=sitePatternCounts_inputfile.csv)</code> A string that will be name of the <code>.csv</code> file <br/></li><li><code>showProgress   (default=true)</code> A boolean argument for visualizing the progress of alignment parsing process<br/></li><li><code>tdigits        (default=3)</code> Number of decimal points to record timme taken to parse the file in seconds<br/></li><li><code>checkpoint     (default=false)</code> A boolean argument to store Phylip object as a .ckp file. (Warning: .ckp can be large).</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/readPhylip.jl#L4-L26">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.show_sp" href="#PhyNEST.show_sp"><code>PhyNEST.show_sp</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">show_sp(p::Phylip)</code></pre><p>Pretty name for displaying all quartet and the corresponding site pattern frequencies on screen in the DataFrame format. <br/>It may result in excessively long table when there are many sequences in the input file.</p><p><strong>Input</strong></p><p><code>p</code>     Phylip object [mandatory]</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/readPhylip.jl#L355-L363">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.write_sp" href="#PhyNEST.write_sp"><code>PhyNEST.write_sp</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">write_sp(p::Phylip)</code></pre><p>Pretty name for exporting all quartet and the corresponding site pattern frequencies in a .csv format. <br/></p><p><strong>Input</strong></p><p><code>p</code>         Phylip object [mandatory] <code>csvname</code>   Filename for the .csv output can be given, otherwise we use PhyNEST_sp.csv by default.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/readPhylip.jl#L411-L419">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.storeCheckPoint" href="#PhyNEST.storeCheckPoint"><code>PhyNEST.storeCheckPoint</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">storeCheckPoint(p::Phylip)</code></pre><p>Stores the phylip object into a .ckp file so a user does not have to repeat the phylip parsing again if working with the <br/>same dataset next time. By default it creates a .ckp file that has the same filename as the input phylip file.<br/>File size can get pretty large... </p><p><strong>Input</strong></p><p><code>p</code>         Phylip object [mandatory]</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/readPhylip.jl#L450-L459">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.readCheckPoint" href="#PhyNEST.readCheckPoint"><code>PhyNEST.readCheckPoint</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">readCheckPoint(ckpfile::AbstractString)</code></pre><p>Reads in .ckp file and creates a filled in phylip object.</p><p><strong>Input</strong></p><p><code>ckpfile</code>   Name of the checkpoint file<br/></p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/readPhylip.jl#L502-L509">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.GetTrueProbsSymm" href="#PhyNEST.GetTrueProbsSymm"><code>PhyNEST.GetTrueProbsSymm</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">GetTrueProbsSymm(myt1::Float64,myt2::Float64,myt3::Float64,theta::Float64)
GetTrueProbsSymm(myt1::Float64,myt2::Float64,myt3::Float64,theta::Float64,alpha::Float64)</code></pre><p>Computes true site pattern probabilities for the symmetric quartet tree, ((1,2),(3,4));. <br/>Three speciation times (or node ages) in coalescent unit and theta must be provided. Alhpa is not <br/>an essential argument for the function and if not provided, it isssumed to be 4/3 by default. <br/>See manuscript or Chifman and Kubatko (2015)[10.1016/j.jtbi.2015.03.006] for more information.<br/></p><p><strong>Input</strong></p><p><code>myt1</code>      Speciation time for the common ancestor of species 1 and 2<br/><code>myt2</code>      Speciation time for the common ancestor of species 3 and 4<br/><code>myt3</code>      Root node age<br/><code>theta</code>     Effective population size parameter<br/><code>alpha</code>     4/3 by default</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Probabilities.jl#L4-L19">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.GetTrueProbsAsymm" href="#PhyNEST.GetTrueProbsAsymm"><code>PhyNEST.GetTrueProbsAsymm</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">GetTrueProbsAsymm(myt1::Float64,myt2::Float64,myt3::Float64,theta::Float64)
GetTrueProbsAsymm(myt1::Float64,myt2::Float64,myt3::Float64,theta::Float64,alpha::Float64)</code></pre><p>Computes true site pattern probabilities for the asymmetric quartet tree, (1,(2,(3,4)));. <br/>Three speciation times (or node ages) in coalescent unit and theta must be provided. Alhpa is not<br/>an essential argument for the function and if not provided, it isssumed to be 4/3 by default. <br/>See manuscript or Chifman and Kubatko (2015)[10.1016/j.jtbi.2015.03.006] for more information.<br/></p><p><strong>Input</strong></p><p><code>myt1</code>      Speciation time for the common ancestor of species 3 and 4<br/><code>myt2</code>      Speciation time for the common ancestor of species 2, 3 and 4<br/><code>myt3</code>      Root node age<br/><code>theta</code>     Effective population size parameter<br/><code>alpha</code>     4/3 by default</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Probabilities.jl#L114-L129">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.GetTrueProbsAsymmTypes" href="#PhyNEST.GetTrueProbsAsymmTypes"><code>PhyNEST.GetTrueProbsAsymmTypes</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">GetTrueProbsAsymmTypes(type::Integer,myt1::Float64,myt2::Float64,myt3::Float64,theta::Float64)
GetTrueProbsAsymmTypes(type::Integer,myt1::Float64,myt2::Float64,myt3::Float64,theta::Float64,alpha::Float64)</code></pre><p>Computes true site pattern probabilities for any of the four the asymmetric quartet trees: <br/></p><ul><li>Type 1: (i,((j,k),l));</li><li>Type 2: ((i,(j,k)),l); </li><li>Type 3: (i,(j,(k,l))); </li><li>Type 4: (((i,j),k),l).<br/></li></ul><p>Three speciation times (or node ages) in coalescent unit and theta must be provided. <br/>Alhpa is not an essential argument for the function and if not provided, it isssumed to be 4/3 by default. <br/>See manuscript or Chifman and Kubatko (2015)[10.1016/j.jtbi.2015.03.006] for more information.<br/></p><p><strong>Input</strong></p><p><code>myt1</code>      Speciation time for the common ancestor of species 3 and 4<br/><code>myt2</code>      Speciation time for the common ancestor of species 2, 3 and 4<br/><code>myt3</code>      Root node age<br/><code>theta</code>     Effective population size parameter<br/><code>alpha</code>     4/3 by default</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Probabilities.jl#L302-L322">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.sim_sp_counts" href="#PhyNEST.sim_sp_counts"><code>PhyNEST.sim_sp_counts</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">sim_sp_counts(type::Integer,myt1::Float64,myt2::Float64,myt3::Float64)
sim_sp_counts(type::Integer,myt1::Float64,myt2::Float64,myt3::Float64,theta::Float64,alpha::Float64,length::Integer)</code></pre><p>This function simulates the 15 site pattern probabilities for five quartet tree topologies, <br/>one symmetric and four asymmetric quartets: </p><ul><li>Type 0: ((i,j),(k,l));</li><li>Type 1: (i,((j,k),l));</li><li>Type 2: ((i,(j,k)),l); </li><li>Type 3: (i,(j,(k,l))); </li><li>Type 4: (((i,j),k),l).<br/></li></ul><p><strong>Input</strong></p><p><code>type</code>   Specify the type of a quartet (use integer)<br/><code>myt1</code>   Speciation time for the most recent internal tree node. Common ancestor of 1 and 2 in the symmetric case<br/><code>myt2</code>   Speciation time for the internal tree node closer to the root<br/><code>myt3</code>   Root node age<br/><code>theta</code>  Effective population size parameter (default=0.01)<br/><code>alpha</code>  Alpha (default=4/3)<br/><code>length</code> Sequence lengths (default=1000000)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Probabilities.jl#L349-L371">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.get_quartets" href="#PhyNEST.get_quartets"><code>PhyNEST.get_quartets</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">get_quartets(net::HybridNetwork; round_gamma_digits=5::Int64)
get_quartets(net::HybridNetwork, p::Phylip; round_gamma_digits=5::Int64, default_theta=0.0001::Float64)</code></pre><p>Extracts information for every quartet extractable from the topology and store it as a Network object. <br/>When the Phylip object is also provided, we fill more slots including the theta, branch lengths for <br/>eac quartet using the moment estimator, average branch lengths using the estimated branch lengths.</p><p><strong>Input</strong></p><ul><li><code>net</code>                   PhyloNetwork.HybridNetwork object for a topology [mandatory]<br/></li><li><code>p</code>                     Phylip object<br/></li><li><code>round_gamma_digits</code>    Number of digits to show for the inhertiance probability in each quartet<br/></li><li>`default_theta          Theta is being estimated when both topology and data are given, however, <br/>                      when the estimated theta is beyond the expected range (i.e., 0.00001 &lt; theta &lt; 0.1)<br/>                      we use user-specified theta (default=0.0001) to estimate branch lengths using <br/>                      the moment estimator.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Quartets.jl#L5-L21">source</a></section></article><div class="admonition is-warning"><header class="admonition-header">Missing docstring.</header><div class="admonition-body"><p>Missing docstring for <code>printQuartets</code>. Check Documenter&#39;s build log for details.</p></div></div><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.get_start_theta" href="#PhyNEST.get_start_theta"><code>PhyNEST.get_start_theta</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">get_start_theta(N::Network; 
                lower_bound=0.00001::Float64,factor=2.0::Float64,tolerance=0.01::Float64)</code></pre><p>Using the Network object that contains information on quartets extracted from the topology and<br/>site pattern frequencies from the data, estimates the `reasonable&#39; theta value that can be<br/>used as the starting point for the composite likelihood optimization. <br/>Approximate interval where the true theta may lie is estimated through two procedures:<br/>somewhat loose interval that keeps the branch lengths positive from the moment estimator,<br/>and then further tightens the interval using the golden section seach.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Theta.jl#L9-L18">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.do_optimization" href="#PhyNEST.do_optimization"><code>PhyNEST.do_optimization</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">do_optimization(net::HybridNetwork, p::Phylip;
                    lower_bound=0.00001::Float64,factor=2.0::Float64,tolerance=0.01::Float64,
                    number_of_itera=1000::Int64)</code></pre><p>Optimizes the composite likelihood of the network topology given the data.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Optimization.jl#L2-L8">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.hill_climbing" href="#PhyNEST.hill_climbing"><code>PhyNEST.hill_climbing</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">hill_climbing(starting_topology::HybridNetwork,p::Phylip,outgroup::String;
    hmax::Integer,
    maximum_number_of_steps::Integer,
    maximum_number_of_failures::Integer,
    number_of_itera::Integer)</code></pre><p>Executes hill climbing algorithm for the network space search.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Searches.jl#L4-L12">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.simulated_annealing" href="#PhyNEST.simulated_annealing"><code>PhyNEST.simulated_annealing</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">simulated_annealing(starting_topology::HybridNetwork, p::Phylip, outgroup::String, 
    hmax::Integer, 
    number_of_burn_in::Int64,
    maximum_number_of_steps::Integer,
    maximum_number_of_failures::Integer,
    number_of_itera::Integer,
    k::Integer,
    cons::Float64,
    alph::Float64
    )</code></pre><p>Executes simulated algorithm for the network space search.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Searches.jl#L169-L182">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.phyne!" href="#PhyNEST.phyne!"><code>PhyNEST.phyne!</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">phyne!(starting_topology::HybridNetwork,p::Phylip,outgroup::String;
    hmax=1::Integer,
    maximum_number_of_steps=1000::Integer,
    maximum_number_of_failures=100::Integer,
    number_of_itera=1000::Integer,
    number_of_runs=5::Integer,
    do_hill_climbing=true::Bool,
    number_of_burn_in=25::Integer,
    k=10::Integer,
    cons=0.5::Float64,
    alph=0.5::Float64
    )</code></pre><p>phyne! is fine.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Searches.jl#L495-L510">source</a></section></article><h2 id="Types"><a class="docs-heading-anchor" href="#Types">Types</a><a id="Types-1"></a><a class="docs-heading-anchor-permalink" href="#Types" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.Phylip" href="#PhyNEST.Phylip"><code>PhyNEST.Phylip</code></a> — <span class="docstring-category">Type</span></header><section><div><pre><code class="language-julia hljs">Phylip</code></pre><p>Subtype of abstract <code>INPUT</code> type with the following attributes:</p><ul><li><code>filename</code>      Name of the input phylip alignment file<br/></li><li><code>time</code>          Time taken for parsing the input in seconds<br/></li><li><code>numtaxa</code>       Number of taxa given at the first line of the input file<br/></li><li><code>seqleng</code>       Sequence length given at the first line of the input file<br/></li><li><code>nametaxa</code>      Sequence names given in the input file<br/></li><li><code>counttaxa</code>     Unique integer identifier given to each individual in the order of appearance in the input file<br/></li><li><code>allquartet</code>    All combinations of quartets for counttaxa<br/></li><li><code>index</code>         Just convert allquartet elements into arbitrary index numbers<br/></li><li><code>spcounts</code>      Arrays of 15 site pattern frequencies for each quartet in allquartet</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Objects.jl#L13-L27">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.Network" href="#PhyNEST.Network"><code>PhyNEST.Network</code></a> — <span class="docstring-category">Type</span></header><section><div><p>Network</p><p>Subtype of abstract <code>Quartet</code> type with the following attributes:</p><ul><li><code>leafname</code>    Name of the leaves in the order of appearance in the input newick<br/></li><li><code>leafnumber</code>  Assigned number for each leaf when reading in the input newick<br/></li><li><code>gamma</code>       Gamma as appears on the newick, although irrelevant for our computation<br/></li><li><code>theta</code>       Estimated theta using the site pattern frequencies. If Network object is<br/>              created without Phylip, theta is not estimated byt default theta=0.001 is displayed.<br/></li><li><code>quartet</code>     An array of all quartets in the given topology</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Objects.jl#L100-L111">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="PhyNEST.quartets" href="#PhyNEST.quartets"><code>PhyNEST.quartets</code></a> — <span class="docstring-category">Type</span></header><section><div><p>quartets</p><p>Subtype of abstract <code>Quartet</code> type with the following attributes:</p><p><code>number</code>        List of individuals quartets in the order of i,j,k,l<br/><code>displayed_tree</code>nth displayed tree that the quartet was extracted from<br/><code>quartet</code>       List of quartets in the order of i,j,k,l using the leaf numbers in HybridNetwork<br/><code>tquartet</code>      List of quartets in the order of i,j,k,l using the leaf numbers in Phylip<br/><code>gamma</code>         Inhertiance probability information provided in HybridNetwork<br/><code>mspcountsNET</code>  We can directly use this counts for likelihood calculation.<br/><code>mrca</code>          List of common ancesters of two taxa in the order of i and j (ij),ik,il,jk,jl,and kl<br/><code>ntau</code>          Unique number of the taus in the tau used in branchlengths<br/><code>momestlength</code>  Branch length identified for each quartet given the mspcounts using moment estimator<br/><code>average_mom_est_bl</code>         Branch length that is averaged for the entire tree/network - Will get filled later because theta is required<br/><code>symtype</code>       Type of each quartet. It can be either symmetric (type 0) or asymmetric. Asymmetric quartets have four possible topologies:</p><ul><li>Type 1: (i,((j,k),l));</li><li>Type 2: ((i,(j,k)),l); </li><li>Type 3: (i,(j,(k,l))); </li><li>Type 4: (((i,j),k),l).<br/></li></ul><p><code>logLik</code>        It&#39;s just a negative likelihood for the quartet in interest</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/sungsik-kong/PhyNEST.jl/blob/bfc1088c734c1a8b7b4538e0a5977cae9881c377/src/Objects.jl#L61-L82">source</a></section></article><h2 id="Index"><a class="docs-heading-anchor" href="#Index">Index</a><a id="Index-1"></a><a class="docs-heading-anchor-permalink" href="#Index" title="Permalink"></a></h2><ul><li><a href="#PhyNEST.Network"><code>PhyNEST.Network</code></a></li><li><a href="#PhyNEST.Phylip"><code>PhyNEST.Phylip</code></a></li><li><a href="#PhyNEST.quartets"><code>PhyNEST.quartets</code></a></li><li><a href="#PhyNEST.GetTrueProbsAsymm"><code>PhyNEST.GetTrueProbsAsymm</code></a></li><li><a href="#PhyNEST.GetTrueProbsAsymmTypes"><code>PhyNEST.GetTrueProbsAsymmTypes</code></a></li><li><a href="#PhyNEST.GetTrueProbsSymm"><code>PhyNEST.GetTrueProbsSymm</code></a></li><li><a href="#PhyNEST.do_optimization"><code>PhyNEST.do_optimization</code></a></li><li><a href="#PhyNEST.get_quartets"><code>PhyNEST.get_quartets</code></a></li><li><a href="#PhyNEST.get_start_theta"><code>PhyNEST.get_start_theta</code></a></li><li><a href="#PhyNEST.greet"><code>PhyNEST.greet</code></a></li><li><a href="#PhyNEST.hill_climbing"><code>PhyNEST.hill_climbing</code></a></li><li><a href="#PhyNEST.phyne!"><code>PhyNEST.phyne!</code></a></li><li><a href="#PhyNEST.readCheckPoint"><code>PhyNEST.readCheckPoint</code></a></li><li><a href="#PhyNEST.readPhylip"><code>PhyNEST.readPhylip</code></a></li><li><a href="#PhyNEST.show_sp"><code>PhyNEST.show_sp</code></a></li><li><a href="#PhyNEST.sim_sp_counts"><code>PhyNEST.sim_sp_counts</code></a></li><li><a href="#PhyNEST.simulated_annealing"><code>PhyNEST.simulated_annealing</code></a></li><li><a href="#PhyNEST.storeCheckPoint"><code>PhyNEST.storeCheckPoint</code></a></li><li><a href="#PhyNEST.write_sp"><code>PhyNEST.write_sp</code></a></li></ul></article><nav class="docs-footer"><a class="docs-footer-nextpage" href="manual/installation/">Installation »</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 0.27.23 on <span class="colophon-date" title="Tuesday 30 May 2023 15:57">Tuesday 30 May 2023</span>. Using Julia version 1.9.0.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
